### PR TITLE
Some suggested changes

### PR DESCRIPTION
**Pull request by sid-kap**

### CODE REVIEW

File: howerpoint.cabal [hunk 1]
```diff
@@ -1,22 +1,15 @@
...
 build-type:           Simple
--- extra-source-files:
 cabal-version:        >=1.10
 
-executable howerpoint
-  main-is:              Howerpoint.hs
-  -- other-modules:
-  -- other-extensions:
-  build-depends:        base >= 4.7 && < 5, deepseq, terminal-size, split
+library
```
Comment (sid-kap) [inline howerpoint.cabal:11]: Made it a library because it has no main function, so cabal was complaining

File: src/Howerpoint.hs [hunk 2]
```diff
@@ -28,30 +28,34 @@ sanitizeCode :: String -> String
...
-  else if isPrefixOf "H>>" line then Just $ drop 3 line
-  else Nothing)
+filterCode = catMaybes . map processLine
+  where
+    processLine :: String -> Maybe String
+    processLine line
+      | isPrefixOf ">>>" line = Just $ drop 3 line
+      | isPrefixOf "L>>" line = Just $ "let " ++ drop 3 line
+      | isPrefixOf "H>>" line = Just $ drop 3 line
+      | otherwise             = Nothing
```
Comment (sid-kap) [inline src/Howerpoint.hs:38]: guards look nicer, easier to read

File: src/Howerpoint.hs [hunk 3]
```diff
@@ -28,30 +28,34 @@ sanitizeCode :: String -> String
...
 
 isCode :: String -> Bool
-isCode = liftA2 (||) (isPrefixOf ">>>") (isPrefixOf "L>>")
+isCode s = isPrefixOf ">>>" s || isPrefixOf "L>>" s
```
Comment (sid-kap) [inline src/Howerpoint.hs:41]: liftA2 is a bit overkill here, in my opinion

File: src/Howerpoint.hs [hunk 4]
```diff
@@ -28,30 +28,34 @@ sanitizeCode :: String -> String
...
 
+isHiddenCode :: String -> Bool
```
Comment (sid-kap) [inline src/Howerpoint.hs:43]: It's nice to have type declarations for all top-level functions

File: src/Howerpoint.hs [hunk 5]
```diff
@@ -28,30 +28,34 @@ sanitizeCode :: String -> String
...
 
 codeFromAllPreviousSlides :: IO String
 codeFromAllPreviousSlides = undefined
 
-help = (sequence $ map (putStrLn . fst . colorize)
- ["\\k\\WTo load test presentation run \\Y:loadTestPresentation",
-  "\\k\\WTo load your own presentation run \\Y:loadPresentation <PATH-TO-YOUR-PRESENTATION>"])
-  >> pure ()
+help :: IO ()
+help = mapM_ (putStrLn . fst . colorize)
```
Comment (sid-kap) [inline src/Howerpoint.hs:56]: I believe `mapM_` is the function you were looking for here. It's equivalent to `void . sequence . map` (where `void :: m a -> m ()` discards the return value of `m a`)

File: src/Howerpoint.hs [hunk 6]
```diff
@@ -85,24 +89,38 @@ moveSlide ps f = do
   displaySlide ps
 
 getTerminalSize :: IO (Int, Int)
-getTerminalSize = T.size >>= (\s -> pure $ maybe (50, 20) (\w -> (T.width w, T.height w)) s)
+getTerminalSize = do
+  s <- T.size
+  return $ maybe (50, 20) (\w -> (T.width w, T.height w)) s
```
Comment (sid-kap) [inline src/Howerpoint.hs:94]: Do-notation looks a bit nicer here

File: src/Howerpoint.hs [hunk 7]
```diff
@@ -120,49 +138,61 @@ padding :: Int -> Int -> (Int, Int)
 padding size contentSize = (half, if odd (size - contentSize) then half + 1 else half)
   where half = floor $ (fromIntegral (size - contentSize)) / 2
 
-repeatN n = take n . repeat
```
Comment (sid-kap) [inline src/Howerpoint.hs:123]: This is actually a library function called `replicate`!

File: src/Howerpoint.hs [hunk 8]
```diff
@@ -120,49 +138,61 @@ padding :: Int -> Int -> (Int, Int)
...
+colorReset :: String
 colorReset = "\x1b[0m"
 
 -- it also takes care of escaping to ;\\k' will be just '\k' and not black color
 colorize :: String -> (String, Int)
-colorize line = if elem '\\' line then colorize' line [] (length line) else (line, length line)
+colorize line =
+  if elem '\\' line
+    then colorize' line [] (length line)
+    else (line, length line)
```
Comment (sid-kap) [inline src/Howerpoint.hs:191]: I find `if/then/else` broken onto 3 lines easier to read, unless it's a really short if/then/else. This is personal preference.

**Comment by sid-kap** [comment]
Some of the changes are my personal style preferences, so you can let me know which of those you don't like, and I can remove them!

File: .ghci [hunk 9]
```diff
@@ -6,6 +6,6 @@
 
 :def loadPresentation \p -> return $ "Presentation {..} <- loadPresentation \"" ++ p ++ "\""
 
-:def loadTestPresentation const . return $ "Presentation {..} <- loadPresentation \"/Users/Frantisek/haskell/howerpoint/test/resources/howerpoint.txt\""
+:def loadTestPresentation const . return $ "Presentation {..} <- loadPresentation \"test/resources/howerpoint.txt\""
```
Comment (sid-kap) [inline .ghci:9]: specifying relative-path works, and has the additional benefit that it works on other peoples' computers

File: src/Howerpoint.hs (hunk 2 above)
Comment (fokot) [inline src/Howerpoint.hs:38]: it looks way better

File: src/Howerpoint.hs (hunk 5 above)
Comment (fokot) [inline src/Howerpoint.hs:56]: Great!

**Comment by fokot** [comment]
I think all of the changes were changes to better. I even learned better coding style from it. Thanks a lot

**Comment by sid-kap** [comment]
You're welcome!

**Pull request merged**